dialogue-core/src/main/java/com/palantir/dialogue/core/StickyEndpointChannels2.java

```diff
@@ -132,6 +135,10 @@ public String toString() {
 
         private final InFlightCallSuccessTransformer successTransformer = new InFlightCallSuccessTransformer();
         private final InFlightCallFailureTransformer failureTransformer = new InFlightCallFailureTransformer();
+        private final AsyncFunction<Throwable, Response> cancellationTransformer = _input -> {
```

Comment:
It's important that we return `Futures.immediateCancelledFuture()` instead of `immediateFailedFuture(throwable)` in this case, otherwise the final future will be failed(CancellationException) (`get()` throws `ExecutionException` caused by `CancellationException` instead of a `CancellationException`).

Reply:
Ah missed that